src/pages/abs/[id]/similar.tsx

```diff
@@ -55,14 +57,16 @@ export const getServerSideProps: GetServerSideProps<ICitationsPageProps> = async
   const result = await adsapi.search.query(params);
   const originalDoc = await getOriginalDoc(adsapi, query.id);
 
-  if (result.isErr()) {
-    return { props: { docs: [], originalDoc, error: result.error.message } };
-  }
-
-  return {
-    props: {
-      docs: result.value.docs,
-      originalDoc,
-    },
-  };
+  return !originalDoc
+    ? { props: { error: 'Document not found' } }
```

Comment:
Looks like we can shortcut this a bit by doing this: https://nextjs.org/blog/next-10#notfound-support

`{ notFound: true }` <-- will automatically show the 404 error page

Reply:
We'll want to check on what kind of error we're getting and act accordingly